cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/max_recent_queries.spec.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  DATASOURCE_NAME,
+  INDEX_PATTERN_WITH_TIME,
+  INDEX_WITH_TIME_1,
+} from '../../../../../utils/apps/constants';
+import { SECONDARY_ENGINE, BASE_PATH } from '../../../../../utils/constants';
+import { NEW_SEARCH_BUTTON } from '../../../../../utils/dashboards/data_explorer/elements.js';
+import {
+  getRandomizedWorkspaceName,
+  setDatePickerDatesAndSearchIfRelevant,
+  generateAllTestConfigurations,
+} from '../../../../../utils/apps/query_enhancements/shared';
+import {
+  generateMaxRecentQueriesTestConfiguration,
+  BaseQuery,
+  TestQueries,
+} from '../../../../../utils/apps/query_enhancements/max_recent_queries';
+
+const workspace = getRandomizedWorkspaceName();
+
+describe('filter for value spec', { testIsolation: true }, () => {
+  const index = INDEX_PATTERN_WITH_TIME.replace('*', '');
+  beforeEach(() => {
+    // Load test data
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data_logs_1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data_logs_1/data_logs_small_time_1.data.ndjson']
+    );
+
+    // Add data source
+    cy.addDataSource({
+      name: DATASOURCE_NAME,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+    // Create workspace
+    cy.deleteWorkspaceByName(workspace);
+    cy.visit('/app/home');
+    cy.osd.createInitialWorkspaceWithDataSource(DATASOURCE_NAME, workspace);
+    cy.createWorkspaceIndexPatterns({
+      workspaceName: workspace,
+      indexPattern: index,
+      timefieldName: 'timestamp',
+      indexPatternHasTimefield: true,
+      dataSource: DATASOURCE_NAME,
+      isEnhancement: true,
+    });
+
+    cy.navigateToWorkSpaceSpecificPage({
+      url: BASE_PATH,
+      workspaceName: workspace,
+      page: 'discover',
+      isEnhancement: true,
+    });
+    cy.getElementByTestId(NEW_SEARCH_BUTTON).click();
+  });
+
+  afterEach(() => {
+    cy.deleteWorkspaceByName(workspace);
+    cy.deleteDataSourceByName(DATASOURCE_NAME);
+    // TODO: Modify deleteIndex to handle an array of index and remove hard code
+    cy.deleteIndex(INDEX_WITH_TIME_1);
+  });
+
+  generateAllTestConfigurations(generateMaxRecentQueriesTestConfiguration).forEach((config) => {
+    if (config.language !== 'PPL' && config.language !== 'OpenSearch SQL') return;
```

Comment:
Could you please leave a comment with why this is the case?

Reply:
I suggest not directly use generateAllTestConfigurations and avoid using if in the permutation. Can we define a permutation just for recent query?

One reference is https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9322/files
For autocomplete, sometimes it only test DQL/SQL as well.

Reply:
okay I saw you defined generateMaxRecentQueriesTestConfiguration why not use that?